### PR TITLE
Improve security for queue settings

### PR DESCRIPTION
**Pull request by james9909**
The queue settings page can now only be viewed by course staff or queue admins, and will 403 otherwise. 

I also noticed `admissionControlUrl` is not being filtered out on queue GET requests - is this something we want to hide from regular users?

### CODE REVIEW

**Comment by nwalters512** [comment]
Thinking about how you chose to exclude the queue props, I think it'd be better to exclude it from the [default Sequelize scope](http://docs.sequelizejs.com/manual/scopes.html) and then only explicitly include it for admins/course staff. That'll give us a sane default for whenever we have to read a queue out of the DB.

**Comment by james9909** [comment]
How would this work when we need the `courseId` for the queue to check for their permissions? Is there an easy way to add an attribute back to the model after it's already been queried? I don't think we should have to make two separate queries to accomplish this.
Also, would this also require us to query each queue with the correct scope in the "Get all open queues" endpoint?

**Comment by nwalters512** [comment]
By this I mean we'd only exclude the private config (admission control) by default; everything else would still be in there. Scopes apply to queries of whole collections too, so nothing would change for the "get 1" vs "get N" cases.

**Comment by james9909** [comment]
Regarding my "get N" question, my confusion was how we'd inject the private config for only the correct queues with one single query. Although I realize now that the queue home page doesn't need that kind of information anyways.

The main thing I'm confused about is how we're supposed to apply the correct "admin" scope when making the queue query without already having queried the queue to see if the user is on course staff, which determines whether they're allowed to see the config or not.

**Comment by nwalters512** [comment]
Ah, I see what you're getting at now. Yeah, I would say for a list, it's fine if we just unconditionally omit them for now. Regarding checking for course staff, that would indeed be not ideal, but DB queries are also super cheap - most requests are already doing several.

I do like the idea of a settings endpoint - if we start adding more settings in the future, that'd be nice to only ever deliver them on the settings page. When I was first doing this, I actually thought about having a separate table for these sorts of settings, but ultimately decided that a 1-1 relationship like that felt weird. We could revisit that too.

File: src/api/queues.js [hunk 1]
```diff
@@ -94,7 +87,20 @@ router.get(
   [requireQueue, failIfErrors],
   safeAsync(async (req, res, _next) => {
     const { id: queueId } = res.locals.queue
-    const queueResults = await Queue.findOne({
+    let queueResult = await Queue.findOne({
```
Comment (nwalters512) [inline src/api/queues.js]: We shouldn't have to re-query here - the queue is already accessible via `res.locals.queue` (see the `requireQueue` middleware).

File: src/api/queues.js [hunk 2]
```diff
@@ -94,7 +87,20 @@ router.get(
...
+      where: {
+        id: queueId,
+      },
+      attributes: ['courseId'],
+    })
+
+    // Convert to plain object that we can manipulate/filter/etc. before
```
Comment (nwalters512) [inline src/api/queues.js]: Comment accidentally copy/pasted?

File: src/api/queues.js [hunk 3]
```diff
@@ -33,18 +33,11 @@ function validateLocation(req, res, next) {
...
-      const queue = queueResult.get({ plain: true })
-      if (isUserStudent(userAuthz, queue.courseId)) {
-        Queue.privateAttributes.forEach(attr => {
-          delete queue[attr]
-        })
-      }
-      return queue
-    })
+    const queuesResult = await Queue.scope(
+      'defaultScope',
```
Comment (nwalters512) [inline src/api/queues.js]: Is specifying `defaultScope` explicitly required?

File: src/api/questions.js [hunk 4]
```diff
@@ -137,17 +137,21 @@ router.post(
...
-    const question = Question.build({
-      name: data.name,
-      // Questions in fixed-location queues should never have a location
-      location: res.locals.queue.fixedLocation ? '' : data.location,
-      topic: data.topic,
-      enqueueTime: new Date(),
-      queueId,
-      askedById: askerId,
-    })
+    const question = await Question.create(
```
Comment (nwalters512) [inline src/api/questions.js]: Can you clarify why there changes were necessary?

File: src/api/queues.js [hunk 5]
```diff
@@ -113,13 +113,25 @@ router.get(
           },
           required: false,
           include: [User],
+          attributes: ['startTime', 'endTime'],
```
Comment (nwalters512) [inline src/api/queues.js]: Can you clarify why the changes in this file were necessary?

File: src/models/index.js [hunk 6]
```diff
@@ -38,8 +38,6 @@ module.exports.initSequelize = sequelize => {
 }
 
 const sequelizeConfig = {
-  // See http://docs.sequelizejs.com/manual/querying.html#operators-security
```
Comment (nwalters512) [inline src/models/index.js]: Can you clarify why this change was necessary?

File: src/api/queues.js (hunk 3 above)
Comment (james9909) [reply src/api/queues.js]: We need `defaultScope` so that the private attribute filtering gets applied, otherwise only `questionCount` will be used.

File: src/api/queues.js (hunk 5 above)
Comment (james9909) [reply src/api/queues.js]: I can't find the exact change from v5, but without `attributes`, `sequelize` will throw an error stating `include.attributes.map is not a function`.

File: src/api/questions.js (hunk 4 above)
Comment (james9909) [reply src/api/questions.js]: We expect that the object returned from `Question.build` contains the appropriate askedBy model through association, but it doesn't do that anymore.
The [docs](http://docs.sequelizejs.com/manual/associations.html#creating-with-associations) suggest to use `.create` to create with association.

File: src/models/index.js (hunk 6 above)
Comment (james9909) [reply src/models/index.js]: For security purposes, string operators (`$or`, `$and`, etc) have been deprecated and `sequelize` will now show deprecation warnings, even though we're not using string operators anywhere.
I was reading through https://github.com/sequelize/sequelize/issues/8417 and it seems like the general consensus was to set `operatorsAliases` to `false`, but that will also trigger a warning as seen [here](https://github.com/sequelize/sequelize/blob/master/lib/sequelize.js#L244-L249).
Removing the option altogether has the same effect as setting it to `false`, but without the warning.

File: src/api/questions.js (hunk 4 above)
Comment (james9909) [reply src/api/questions.js]: Looking through the docs some more, it's possible to use `.build()` to achieve the same effect, but since we're not doing anything special between `.build()` and `.save()`, I don't see a reason to use it over `.create()`
